compression/deflate_writer.mbt

```diff
@@ -0,0 +1,103 @@
+///|
+/// DeflateWriter wraps an io.Writer and compresses data using DEFLATE on write.
+/// Similar to Java's DeflateOutputStream.
+pub struct DeflateWriter {
+  sink : &@io.Writer
+  bit_writer : BitWriter
+  priv mut closed : Bool
+}
+
+///|
+/// Create a new DeflateWriter that writes compressed data to the sink.
+pub fn DeflateWriter::new(sink : &@io.Writer) -> DeflateWriter {
+  DeflateWriter::{
+    sink,
+    bit_writer: bit_writer(),
+    closed: false,
+  }
+}
+
+///|
+/// Write uncompressed bytes, compressing them before writing to the underlying writer.
+/// Returns the number of uncompressed bytes accepted (not the compressed bytes written).
+pub impl @io.Writer for DeflateWriter with write(
+  self,
+  arr : FixedArray[Byte],
+  off : Int,
+  len : Int
+) -> Int {
+  if self.closed || len <= 0 {
+    return 0
+  }
+
+  // TODO: Implement actual DEFLATE compression here
```

Comment:
Isn't the deflate algorithm already implemented in feat-deflate? please implement the logics here.